microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/ls/ApplicationPropertiesTextDocumentService.java

```diff
@@ -243,7 +243,7 @@ public void didSave(DidSaveTextDocumentParams params) {
 		MicroProfileProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument());
 		return getProjectInfoCache().getProjectInfo(projectInfoParams).thenComposeAsync(projectInfo -> {
 			if (projectInfo.getProperties().isEmpty()) {
-				return null;
+				return CompletableFuture.completedFuture(null);
```

Comment:
You did that to avoid NPE in thenApply(info

Reply:
I noticed that `return null` also caused an NPE so I made this change.

Where is the `thenApply(info)` you are referring to?

Is `thenApply(info)` being called by the caller for `public CompletableFuture<List<Either<Command, CodeAction>>> codeAction(CodeActionParams params)`?

Reply:
Forget my comments.